projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_base.py

```diff
@@ -521,18 +547,17 @@ def _payload_poster_thread(self):
             finally:
                 self._payloads_queue.task_done()
 
-            # Complete Post-Ingestion operations
-            try:
-                self._ingester.post_ingest_process(
-                    payload=payload_dict,
+            # If there are any post-processors set, complete the post-process
+            # operations
+            if self._post_processors:
+                self._execute_post_process_operations(
+                    payload=payload_obj,
```

Comment:
I was looking into post processor implementation, and did notice the post_ingest_process() payload parameter is optional? why is that?

Reply:
Refactored.